### PR TITLE
displaylink: changed systemd start to not block

### PR DESCRIPTION
**Pull request by eyJhb**
Previously during boot, displaylink would hang during boot for the 2 minutes timeout.
This fixes issue #51684 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/linux/displaylink/default.nix [hunk 1]
```diff
@@ -44,6 +44,7 @@ in stdenv.mkDerivation rec {
       main systemd "$out/lib/udev/rules.d/99-displaylink.rules" "$out/lib/udev/displaylink.sh"
     )
     sed -i '2iPATH=${systemd}/bin:$PATH' $out/lib/udev/displaylink.sh
+    sed -i 's,systemctl start,systemctl start --no-block,g' $out/lib/udev/displaylink.sh
```
Comment (Mic92) [inline pkgs/os-specific/linux/displaylink/default.nix]: Can you replace both `sed` commands by a proper patch that, uses substituteAll for the dynamic parts? https://github.com/NixOS/nixpkgs/blob/c19083961d4bb271d9aa63d61f855683137bba4b/pkgs/development/python-modules/pyocr/default.nix#L21

**Comment by eyJhb** [comment]
Well, the first sed command inserts a line onto line 2, so don't know how I
would substitute for that. And the secound just replaces some text, so it
could easily be replaced by substitute, but not substituteAll, as the file
operating on, is downloaded from a external source (no templating
variables) :/

Have I understood correctly?

**Comment by Mic92** [comment]
I mean one could build a patch that places `PATH=@systemd@/bin:$PATH` on the second line and the replaces the systemctl command that is also in the same file.

File: pkgs/os-specific/linux/displaylink/udev-installer.patch [hunk 2]
```diff
@@ -0,0 +1,18 @@
+--- a/udev-installer.sh	2018-12-09 12:05:53.772318942 +0100
++++ b/udev-installer.sh	2018-12-09 12:06:19.939947629 +0100
+@@ -21,12 +21,12 @@
+   cat <<'EOF'
+ start_service()
+ {
+-  systemctl start dlm
++  @systemd@/bin/systemctl start dlm
```
[the remark this answers is not in the document]
Comment (eyJhb) [reply pkgs/os-specific/linux/displaylink/udev-installer.patch]: Fixed, thanks for pointing it out :)

**Pull request merged**